gprofiler/docker_client.py

```diff
@@ -59,12 +59,21 @@ def _safely_get_process_container_name(self, pid: int) -> Optional[str]:
             logger.warning(f'Could not get a container name for PID {pid}', exc_info=True)
             return None
 
-    @functools.lru_cache(4096)
-    def _get_container_name(self, container_id):
-        container = self._client.containers.get(container_id)
-        container_name = container.name
-        self._current_container_names.add(container_name)
-        return container_name
+    def _get_container_name(self, container_id) -> Optional[str]:
+        if container_id in self._container_id_to_name_cache:
+            return self._container_id_to_name_cache[container_id]
+
+        self.refresh_container_names_cache()
```

Comment:
I would not cause a `cache.clear()` here (when calling `refresh_container_names_cache`) and only append to the list of known container-ids, since a container might die during this and I would still want its name :)

Reply:
If a container died here, we won't see its ID in `/proc/pid/cgroup`. The only chance for us to see a container ID who's eliminated from the cache here is if it died between the time we read its `/proc/pid/cgroup` to the time we tried to get its name by ID. Pretty rare race IMO.